client/indexer-service/src/handler.rs

```diff
@@ -361,8 +382,20 @@ impl IndexerService {
                 old_capacity: _old_capacity,
                 next_block_when_change_allowed: _next_block_when_change_allowed,
             } => match provider_id {
-                StorageProviderId::BackupStorageProvider(_) => {
+                StorageProviderId::BackupStorageProvider(bsp_id) => {
                     Bsp::update_capacity(conn, who.to_string(), new_capacity.into()).await?;
+
+                    // update also the stake
+                    let current_block_hash = self.client.info().best_hash;
```

Comment:
This most often will not be the correct `current_block_hash` - as in the block we are currently processing.

You should pass and use the `block_hash` parameter from `index_block` method.